### PR TITLE
New Activity for DTOSampleApp Automation

### PR DESCRIPTION
**Pull request by ramanamca04**
New Activity created for DTOSampleApp Automation, that allows to input EmbedCd, Pcode, ApiKey and SecretKey from SampleApp.

### CODE REVIEW

File: PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/players/OoyalaSkinOPTPlayerActivity.java [hunk 1]
```diff
@@ -26,20 +26,30 @@
 import java.util.Observable;
 import java.util.Observer;
 
+//added below two imports by rmanchi
```
Comment (akshaydch11) [inline PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/players/OoyalaSkinOPTPlayerActivity.java]: remove this comment

File: PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/players/OoyalaSkinOPTPlayerActivity.java [hunk 2]
```diff
@@ -26,20 +26,30 @@
...
+  private static final int PERMISSIONS_REQUEST_WRITE_EXTERNAL_STORAGE = 1;
+  boolean writePermission = false;
 
   String EMBED = null;
   String PCODE  = null;
   String DOMAIN = null;
 
-  private final String APIKEY = "";
-  private final String SECRET = "";
+  //Changed by rmanchi - Removed final keyword for apikey, secret & account_id
```
Comment (akshaydch11) [inline PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/players/OoyalaSkinOPTPlayerActivity.java]: remove this comment

File: PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/players/OoyalaSkinOPTPlayerActivity.java [hunk 3]
```diff
@@ -193,6 +215,14 @@ public void update(Observable arg0, Object argN) {
     Playbacklog.writeToSdcardLog(text);
 
     Log.d(TAG, "Notification Received: " + arg1 + " - state: " + player.getState());
+
+    //added by rmanchi
```
Comment (akshaydch11) [inline PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/players/OoyalaSkinOPTPlayerActivity.java]: remove this comment

File: PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/utils/PlayerSelectionOption.java [hunk 4]
```diff
@@ -11,14 +11,52 @@
   private Class <? extends Activity> activity;
   private String pcode;
   private String domain;
+  //added by rmanchi
```
Comment (akshaydch11) [inline PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/utils/PlayerSelectionOption.java]: remove comment

File: PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/utils/PlayerSelectionOption.java [hunk 5]
```diff
@@ -11,14 +11,52 @@
...
+  private String accountId;
 
-  public PlayerSelectionOption(String embedCode, String pcode, String domain, Class<? extends Activity> activity) {
+  public PlayerSelectionOption(String embedCode, String pcode, String apiKey, String secretKey, String accountId, String domain, Class<? extends Activity> activity) {
     this.embedCode = embedCode;
     this.activity = activity;
     this.pcode = pcode;
     this.domain = domain;
+
+    //added by rmanchi
```
Comment (akshaydch11) [inline PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/utils/PlayerSelectionOption.java]: remove comment

File: PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/utils/PlayerSelectionOption.java [hunk 6]
```diff
@@ -11,14 +11,52 @@
...
+    this.apiKey = apiKey;
+    this.secretKey = secretKey;
+    this.accountId = accountId;
+
+  }
+
+  //added by rmanchi
```
Comment (akshaydch11) [inline PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/utils/PlayerSelectionOption.java]: remove comment

File: PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/utils/PlayerSelectionOption.java [hunk 7]
```diff
@@ -11,14 +11,52 @@
...
+  /**
+   * Get the pcode for this sample
+   * @return the apikey
+   */
+  public String getApiKey() {
+    return apiKey;
+  }
+
+  //added by rmanchi
```
Comment (akshaydch11) [inline PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/utils/PlayerSelectionOption.java]: remove comment

File: PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/utils/PlayerSelectionOption.java [hunk 8]
```diff
@@ -11,14 +11,52 @@
...
+  /**
+   * Get the pcode for this sample
+   * @return the secretKey
+   */
+  public String getSecretKey() {
+    return secretKey;
   }
 
+  //added by rmanchi
```
Comment (akshaydch11) [inline PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/utils/PlayerSelectionOption.java]: remove comment

File: PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/players/OoyalaSkinOPTPlayerActivity.java [hunk 9]
```diff
@@ -52,11 +62,23 @@
...
+    } else {
+      writePermission= true;
+    }
+
     setTitle(getIntent().getExtras().getString("selection_name"));
     setContentView(R.layout.player_skin_simple_layout);
     EMBED = getIntent().getExtras().getString("embed_code");
     PCODE = getIntent().getExtras().getString("pcode");
     DOMAIN = getIntent().getExtras().getString("domain");
+    //added by rmanchi
```
Comment (akshaydch11) [inline PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/players/OoyalaSkinOPTPlayerActivity.java]: remove comment

File: PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/players/OoyalaSkinOPTPlayerActivity.java [hunk 10]
```diff
@@ -52,11 +62,23 @@
   @Override
   public void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
+    //added by rmanchi
```
Comment (akshaydch11) [inline PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/players/OoyalaSkinOPTPlayerActivity.java]: remove comment

File: PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/players/OoyalaSkinOPTPlayerActivity.java [hunk 11]
```diff
@@ -26,20 +26,30 @@
...
+import static android.Manifest.permission.WRITE_EXTERNAL_STORAGE;
+import static android.content.pm.PackageManager.PERMISSION_GRANTED;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+
 public class OoyalaSkinOPTPlayerActivity extends Activity
   implements Observer, DefaultHardwareBackBtnHandler, EmbedTokenGenerator {
 
   private static final String TAG = OoyalaSkinOPTPlayerActivity.class.getSimpleName();
+  //added below two stmts by rmanchi
```
Comment (akshaydch11) [inline PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/players/OoyalaSkinOPTPlayerActivity.java]: remove comment

File: PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/lists/BasicPlaybackListActivity.java [hunk 12]
```diff
@@ -41,9 +41,25 @@ public void onCreate(Bundle savedInstanceState) {
 
     selectionMap = new LinkedHashMap<String, PlayerSelectionOption>();
     //Populate the embed map
-    selectionMap.put("Widevine DASH Stream", new PlayerSelectionOption("Q1cG85NTE6Df3A95XMMbKGsPg6yaEZGm","BjcWYyOu1KK2DiKOkF41Z2k0X57l", "http://ooyala.com", OoyalaSkinOPTPlayerActivity.class));
-    selectionMap.put("Widevine DASH Downloader", new PlayerSelectionOption("Q1cG85NTE6Df3A95XMMbKGsPg6yaEZGm","BjcWYyOu1KK2DiKOkF41Z2k0X57l", "http://ooyala.com", OfflineDownloadActivity.class));
-    selectionMap.put("Widevine DASH Offline Player", new PlayerSelectionOption("Q1cG85NTE6Df3A95XMMbKGsPg6yaEZGm","BjcWYyOu1KK2DiKOkF41Z2k0X57l", "http://ooyala.com", OfflineSkinPlayerActivity.class));
+
+    //added by rmanchi - IF condition
```
Comment (akshaydch11) [inline PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/lists/BasicPlaybackListActivity.java]: remove comment

File: PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/complete/MainExoPlayerActivity.java [hunk 13]
```diff
@@ -47,6 +47,9 @@ protected void onCreate(Bundle savedInstanceState) {
     activityMap.put(BasicPlaybackListActivity.getName(), BasicPlaybackListActivity.class);
     activityMap.put(DownloadSerializationActivity.getName(), DownloadSerializationActivity.class);
 
+    //Added by rmanchi
```
Comment (akshaydch11) [inline PlaybackLab/DownloadToOwnSampleApp/app/src/main/java/com/ooyala/sample/complete/MainExoPlayerActivity.java]: remove comment

**Review by akshaydch11 — CHANGES_REQUESTED** [review]
Please remove "added by rmanchi" comments from everywhere.

**Comment by ramanamca04** [comment]
@akshaydch11  Worked on review comments given by you. Could you please verify. Thanks!

**Review by brol1dev — APPROVED** [review]
LGTM

**Comment by akshaydch11** [comment]
@ramanamca04  Please resolve conflicts and update PR.

**Pull request closed without merging**